static/app/components/carousel.tsx

```diff
@@ -0,0 +1,148 @@
+    } else if (scrollLeft >= totalScrollDist) {
+      setIsAtStart(false);
+      setIsAtEnd(true);
+    } else {
+      setIsAtStart(false);
+      setIsAtEnd(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const contentWidth = ref.current.scrollWidth - ref.current.clientWidth;
+    if (contentWidth === 0) {
+      setShowArrows(false);
+    }
+
+    const scrollLeft = ref.current.scrollLeft;
+    setStartAndEnd(scrollLeft, contentWidth);
+  }, [ref]);
+
+  const handleScroll = (direction: string) => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+      const itemWidth = parseInt(getComputedStyle(ref.current.children[0]).width, 10);
+
+      let scrollDist = 0;
+      if (direction === 'left') {
+        scrollDist = scrollLeft - itemWidth;
+      } else if (direction === 'right') {
+        scrollDist = scrollLeft + itemWidth;
+      }
+      ref.current.scrollLeft = scrollDist;
+    });
+  };
+
+  const setArrows = () => {
+    requestAnimationFrame(() => {
+      if (!ref.current) {
+        return;
+      }
+
+      const scrollLeft = ref.current.scrollLeft;
+      const totalScrollDist = ref.current.scrollWidth - ref.current.clientWidth;
+
+      setStartAndEnd(scrollLeft, totalScrollDist);
+    });
+  };
+
+  return (
+    <CarouselContainer>
+      <CarouselItems ref={ref} onScroll={setArrows}>
+        {children}
+      </CarouselItems>
+      {showArrows && !isAtStart && (
+        <CircledArrow onClick={() => handleScroll('left')} direction="left" />
+      )}
+      {showArrows && !isAtEnd && (
+        <CircledArrow onClick={() => handleScroll('right')} direction="right" />
+      )}
+    </CarouselContainer>
+  );
+}
+
+const CarouselContainer = styled('div')`
+  position: relative;
+`;
+
+const CarouselItems = styled('div')`
+  display: flex;
+  overflow-x: scroll;
+  scroll-behavior: smooth;
+  &::-webkit-scrollbar {
+    background-color: transparent;
+    height: 8px;
+  }
+  &::-webkit-scrollbar-thumb {
+    background: ${p => p.theme.gray400};
+    border-radius: 8px;
+  }
+`;
+
+type ArrowProps = {
+  direction: 'up' | 'down' | 'left' | 'right';
+  onClick: () => void;
+};
+
+function CircledArrow({onClick, direction}: ArrowProps) {
+  return (
+    <StyledArrowButton
+      onClick={onClick}
+      direction={direction}
+      data-test-id={`arrow-${direction}`}
+    >
+      <IconArrow color="black" size="sm" direction={direction} />
+    </StyledArrowButton>
+  );
+}
+
+const StyledArrowButton = styled(Button)<{direction: string}>`
+  position: absolute;
+  ${p => (p.direction === 'left' ? `left: 0;` : `right: 0;`)}
+  top: 0;
+  bottom: 0;
+  height: 36px;
+  width: 36px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: 66px;
+  border: 1px solid ${p => p.theme.gray200};
+  padding: 0;
+  margin: auto;
+  background-color: white;
```

Comment:
```suggestion
  background-color: ${p => p.theme.white};
```

otherwise this wont' work in dark mode.

Though ideally we're just not changing the color of the button

Reply:
Shouldn't it be `p.theme.background`?